crates/networking/p2p/peer_channels.rs

```diff
@@ -0,0 +1,112 @@
+use std::{sync::Arc, time::Duration};
+
+use ethrex_core::{
+    types::{BlockBody, BlockHeader},
+    H256,
+};
+use tokio::sync::{mpsc, Mutex};
```

Comment:
Are we sure we want to use tokio's mutex here? [Tokio's docs have something about this topic](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use).


Can we double check we actually want to use tokio's mutex? wdyt?

Reply:
Each peer has only one receiver so (in cases with one or few peers) if we have multiple tasks sending requests to the same peer we would have to wait for a previous process to release the receiver's mutex before processing its response to a new request. Tokio's Mutex allows us to await on the mutex's lock. Does this make sense?

Reply:
Yup, makes sense.